src/libraries/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriterAsync.cs

```diff
@@ -73,6 +73,62 @@ internal override Task WriteXmlDeclarationAsync(string xmldecl)
             return Task.CompletedTask;
         }
 
+        protected override async ValueTask DisposeAsyncCore()
+        {
+            try
+            {
+                await FlushBufferAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                // Future calls to Close or Flush shouldn't write to Stream or Writer
+                writeToNull = true;
+
+                if (stream != null)
+                {
+                    try
+                    {
+                        await stream.FlushAsync().ConfigureAwait(false);
+                    }
+                    finally
+                    {
+                        try
+                        {
+                            if (closeOutput)
+                            {
+                                await stream.DisposeAsync().ConfigureAwait(false);
+                            }
+                        }
+                        finally
+                        {
+                            stream = null;
+                        }
+                    }
+                }
+                else if (writer != null)
```

Comment:
should this always be true? if so perhaps just use `else` and add an Assert

Reply:
I think technically that is true, but for whatever reason, if DisposeAsyncCore() have called again or somehow it was already disposed, i think its better to check for null